hy/compiler.py

```diff
@@ -1226,6 +1227,12 @@ def _compile_import(expr, module, names=None, importer=ast.Import):
         while len(expr) > 0:
             iexpr = expr.pop(0)
 
+            if iexpr[0] == "__future__" and "TailRec" in iexpr[1]:
+                self.tail_rec = True
+                TRInd = iexpr[1].index("TailRec")
+                iexpr[1].pop(TRInd)
```

Comment:
Also, it should probably be ensured that `__future__` imports are at the top of the file, just like Python does.